planner/core/plan_to_pb.go

```diff
@@ -78,11 +98,20 @@ func (p *PhysicalSelection) ToPB(ctx sessionctx.Context) (*tipb.Executor, error)
 	selExec := &tipb.Selection{
 		Conditions: conditions,
 	}
-	return &tipb.Executor{Tp: tipb.ExecType_TypeSelection, Selection: selExec}, nil
+	executorID := ""
+	if storeType == kv.TiFlash {
+		var err error
+		selExec.Child, err = p.children[0].ToPB(ctx, storeType)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		executorID = p.ExplainID().String()
+	}
```

Comment:
would them be encapsulated in a function?

Reply:
It's hard, because `selExec` is not an interface.